service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementExtractor.java

```diff
@@ -0,0 +1,128 @@
+import org.hl7.fhir.dstu3.model.RelatedPerson;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.instance.model.api.IIdType;
+
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+
+public class RequestStatementExtractor {
+    private static final String RECIPIENT_PRACTITIONER = "Recipient Practitioner: ";
+    private static final String RECIPIENT_HEALTH_CARE_SERVICE = "Recipient Healthcare Service: ";
+    private static final String RECIPIENT_ORG = "Recipient Org: ";
+    private static final String NOTE_AUTHOR = "Author: ";
+    private static final String NOTE_AUTHOR_RELATION = NOTE_AUTHOR + "Relation ";
+    private static final String NOTE_AUTHOR_PRACTITIONER = NOTE_AUTHOR + "Practitioner ";
+    private static final String NOTE_AUTHOR_PATIENT = NOTE_AUTHOR + "Patient";
+    private static final String COMMA = ",";
+    private static final String NAME = "%s %s";
+
+    public static String extractServiceRequested(ReferralRequest referralRequest) {
+        return referralRequest.getServiceRequested().stream()
+            .map(value -> extractTextOrCoding(value))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.joining(COMMA));
+    }
+
+    public static String extractDevice(Device device) {
+        return extractTextOrCoding(device.getType())
+            .orElse(StringUtils.EMPTY);
+    }
+
+    public static String extractRecipient(MessageContext messageContext, Reference reference) {
+        IIdType referenceId = reference.getReferenceElement();
+        if (referenceId.getResourceType().equals(ResourceType.Practitioner.name())) {
+            return messageContext.getInputBundleHolder()
+                .getResource(referenceId)
+                .map(resource -> (Practitioner) resource)
+                .map(value -> RECIPIENT_PRACTITIONER + extractHumanName(value.getNameFirstRep()))
+                .orElse(StringUtils.EMPTY);
+        } else if (referenceId.getResourceType().equals(ResourceType.HealthcareService.name())) {
+            return messageContext.getInputBundleHolder()
+                .getResource(referenceId)
+                .map(resource -> (HealthcareService) resource)
+                .map(value -> RECIPIENT_HEALTH_CARE_SERVICE + value.getName())
+                .orElse(StringUtils.EMPTY);
+        } else if (referenceId.getResourceType().equals(ResourceType.Organization.name())) {
+            return messageContext.getInputBundleHolder()
+                .getResource(referenceId)
+                .map(resource -> (Organization) resource)
+                .map(value -> RECIPIENT_ORG + value.getName())
+                .orElse(StringUtils.EMPTY);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    public static String extractReasonCode(ReferralRequest referralRequest) {
+        var ignoreFirstReasonCode = referralRequest.getReasonCode().subList(1, referralRequest.getReasonCode().size());
```

Comment:
Could sublist throw ArrayIndexOutOfBounds if there are no reason codes?

It would be safer to use skip(1) on the stream.